src/App.tsx

```diff
@@ -15,36 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+interface SortParam {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getSortedGoods(goods: string[], { sortField, isReversed }: SortParam) {
+  const sortededGoods = [...goods];
+
+  if (sortField) {
+    sortededGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortededGoods.reverse();
+  }
+
+  return sortededGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.default);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
if you set primitive value by default , it's add types under the hood so can skip it